### PR TITLE
[SMP-1656] SMP 0.8.0 external self-managed Redis

### PR DESCRIPTION
**Pull request by brian-f-harness**
SMP external Redis 0.8.0

For reviewers, [a preview](https://64c7c97c19164106a8bf274f--harness-developer.netlify.app/tutorials/self-managed-enterprise-edition/use-an-external-redis-database) is available.

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+Harness recommends a Redis configuration with the following minimum hardware:
+
+- 3 virtual machines
+- 8 cores per machine
+- 8GB memory per machine
+- 64GB disk per machine (SSD preferred)
+- 1GB/s minimum network bandwidth
+- 
+## Software requirements
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md:33]:
```suggestion
- 1GB/s minimum network bandwidth

## Software requirements
```

File: docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+External database setup requires the following software:
+
+- Supported OS (Ubuntu 20.04 LTS)
+- [Redis enterprise](https://redis.com/redis-enterprise-software/download-center/software/)
+
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md:39]:
```suggestion
- Supported OS (Ubuntu 20.04 LTS)
- [Redis enterprise download URL](https://redis.com/redis-enterprise-software/download-center/software/)

```

File: docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+1. Open https://<EXTERNAL_IP_OF_NODE1>:8443 in your browser, and then complete the following on the node setup page. 
+
+   1. Enable Flash Storage.
+
+   2. Select **+** next to the IP, add the external IP, and then select **Save**.
+
+   3. Disable external traffic on the internal IP.
+
+   4. Add the DNS record subdomain in the **FQDN** field, for example `redis.harness-redis.com`.
+
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md:145]:
```suggestion
   4. Add the DNS record subdomain in the **FQDN** field which we used in NS record, for example `redis.harness-redis.com`.

```

File: docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+   5. Create your admin credentials.
+
+   6. Select **Create Cluster**.
+
+   7. Select **Next**.
+
+     :::info
+     You can skip the cluster key if the page opens.
+     :::
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md:154]:
```suggestion
     :::info
     You can skip the cluster key if the page opens for now. (you can procure a license key independently from redislabs)
     :::
```

File: docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+   
+   ```shell
+   set foo bar
+   ```
+
+5. Run the following.
+
+   ```shell
+   get foo
+   ```
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md:245]:
```suggestion
1. Run the following command.

   ```shell
   redis-cli -h <YOUR_INTERNAL_ENDPOINT> -p <port>
   ```
2. Run the following.
  
   ```shell
   ping
   ```

  The expected response is `pong`.

3. Run the following.
   
   ```shell
   set foo bar
   ```

4. Run the following.

   ```shell
   get foo
   ```
```

File: docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+   global:
+     database:
+       redis:
+         installed: false
+         hosts:
+         - <YOUR_INTERNAL_ENDPOINT_AND_PORT>
+         secretName: "YOUR_REDIS_USER_PASSWORD"
+         userKey: "username"
+         passwordKey: "password"
+   ```
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/advanced-configurations/use-an-external-redis-database.md:275]:
```suggestion
To configure your Harness environment and Helm chart, do the following:

1. Add the following override entries to your newly-installed Harness Redis instance.

   ```yaml
   global:
     database:
       redis:
         installed: false
         hosts:
         - <YOUR_INTERNAL_ENDPOINT_AND_PORT>
         secretName: ""
         userKey: ""
         passwordKey: ""
   ```
```

**Review by pratmit — DISMISSED** [review]
LGTM, based on the technical details provided

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+   | ICMP| * | Redis gossip protocol|
+   | UDP| 53, 5353 | Redis DNS|
+   | TCP| 10000-20000 | Redis connectivity|
+   | TCP| 8443 | Secure (HTTPS) access to the web management UI|
+
+4. For testing purposes, keep the Source IP ranges set to 0.0.0.0/0.
+
+## Install Redis
+
+After you configure your firewall rules, you must install Redis. To install Redis, you must create three VMs, create a public zone, configure your Redis cluster, configure your Redis database, and test your connectivity.
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:68]:
```suggestion
After you configure your firewall rules, you must install Redis. To install Redis, you must create three VMs, create DNS records, configure your Redis cluster, configure your Redis database, and test your connectivity.
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+- Supported OS: Ubuntu 20.04 LTS
+
+## Prerequisites
+
+The following prerequisites are needed:
+
+- A Redis Labs account. To sign up, go to the [Redis Cloud Console](https://app.redislabs.com/#/).
+
+- Redis Enterprise. To download Redis for Ubuntu 20.04, go to [Redis enterprise](https://redis.com/redis-enterprise-software/download-center/software/).
+
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:46]:
```suggestion
- Redis Enterprise. To download Redis for Ubuntu 20.04, go to [Redis enterprise](https://redis.com/redis-enterprise-software/download-center/software/).

- Public or Internal domain name. eg harness-redis-abc.com
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+    sudo ./install.sh -y
+    ```
+
+8. Check the installation logs for errors.
+
+9. Repeat the steps to configure your other two VMs.
+
+### Create a public zone
+
+To create a public zone, do the following:
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:111]:
```suggestion
To create DNS records, do the following:
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+2. Open the public zone page. 
+
+3. Create three type A records with subdomains for the public domain name, for example, `node1.harness-redis-abc.com`, `node2.harness-redis-abc.com`, and `node3.harness-redis-abc.com`.
+
+4. Add the internal IP and external IP for each VM in their respective A records.
+
+5. Create one NS record with a different subdomain, for example `redis.harness-redis.com`.
+
+6. Add the subdomains of all your nodes, for example `node1.harness-redis.com`, `node2.harness-redis.com`, and `node3.harness-redis.com`.
+
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:124]:
```suggestion
1. Go to the DNS portal of your domain registrar.

2. Create three type A records with subdomains for the public domain name, for example, `node1.harness-redis-abc.com`, `node2.harness-redis-abc.com`, and `node3.harness-redis-abc.com`.

3. Add the internal IP and external IP for each VM in their respective A records.

4. Create one NS record with a different subdomain, for example `redis.harness-redis.com`.

5. Add the subdomains of all your nodes, for example `node1.harness-redis.com`, `node2.harness-redis.com`, and `node3.harness-redis.com`.

```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+   6. Select **Create Cluster**.
+
+   7. Select **Next**.
+
+     :::info
+     You can skip the cluster key for now if the page opens. (You can procure a license key from Redis Labs later, if required.)
+     :::
+
+3. Open https://<EXTERNAL_IP_OF_NODE2>:8443 in your browser, and then complete the following on the node setup page.
```
Comment (aimurphy) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:149]:
```suggestion
3. Open `https://<EXTERNAL_IP_OF_NODE2>:8443` in your browser, and then complete the following on the node setup page. 
```

File: src/components/LandingPage/data/smpData.tsx [hunk 12]
```diff
@@ -15,4 +15,16 @@ export const SMPList: CardItem[] = [
...
     link: "/tutorials/self-managed-enterprise-edition/use-an-external-mongodb-database",
   },
+  {
+    title: "Configure an external self-managed Redis database",
+    module: MODULES.smp,
+    icon: "img/logo.svg",
+    description: (
+      <>Learn how to use an external self-managed Redis database with Self-Managed Enterprise Edition installations.</>
+    ),
+    newDoc: false,
```
Comment (aimurphy) [inline src/components/LandingPage/data/smpData.tsx:25]:
```suggestion
    newDoc: true,
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 13]
```diff
@@ -0,0 +1,254 @@
+---
+title: Use an external self-managed Redis database with your installation
+description: Learn how to use an external self-managed Redis database with Self-Managed Enterprise Edition installations.
+sidebar_label: Configure an external self-managed Redis database
```
Comment (aimurphy) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:4]:
```suggestion
sidebar_label: External self-managed Redis
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 14]
```diff
@@ -0,0 +1,254 @@
...
+
+   4. Select **Join cluster**.
+
+   5. Add the internal IP of your first node.
+
+   6. Enter your admin credentials.
+
+   7. Select **Create**.
+
+3. Open https://<EXTERNAL_IP_OF_NODE3>:8443 in your browser, and then complete the following on the node setup page.
```
Comment (aimurphy) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:165]:
```suggestion
3. Open `https://<EXTERNAL_IP_OF_NODE3>:8443` in your browser, and then complete the following on the node setup page. 
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 15]
```diff
@@ -0,0 +1,254 @@
...
+
+   6. Enter your admin credentials.
+
+   7. Select **Create**.
+
+### Configure your Redis database
+
+To configure your Redis database, do the following:
+
+1. Open https://<YOUR_NODE1_EXTERNAL_IP>:8443 in your browser.
```
Comment (aimurphy) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:185]:
```suggestion
1. Open `https://<YOUR_NODE1_EXTERNAL_IP>:8443` in your browser.
```

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md [hunk 16]
```diff
@@ -0,0 +1,254 @@
...
+   ```yaml
+   global:
+     database:
+       redis:
+         installed: false
+         hosts:
+         - <YOUR_INTERNAL_ENDPOINT_AND_PORT>
+         secretName: ""
+         userKey: ""
+         passwordKey: ""
```
Comment (aimurphy) [inline tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:254]: It looks like the closing is missing for this code block.

Also I would just double-check the spacing and code blocks in the whole tutorial. On GH, it looks like something is up with the format but I don't see any issue on the preview.

File: tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md (hunk 15 above)
Comment (aimurphy) [reply tutorials/self-managed-enterprise-edition/use-an-external-redis-database.md:185]: One of these sections is duplicated.

**Review by aimurphy — CHANGES_REQUESTED** [review]

File: src/components/LandingPage/data/smpData.tsx [hunk 17]
```diff
@@ -15,4 +15,16 @@ export const SMPList: CardItem[] = [
...
+  {
+    title: "Configure an external self-managed Redis database",
+    module: MODULES.smp,
+    icon: "img/logo.svg",
+    description: (
+      <>Learn how to use an external self-managed Redis database with Self-Managed Enterprise Edition installations.</>
+    ),
+    newDoc: true,
+    type: [docType.Documentation],
+    time: "5 min",
```
Comment (aimurphy) [inline src/components/LandingPage/data/smpData.tsx:27]:
```suggestion
    time: "15 min",
```

**Review by aimurphy — APPROVED** [review]
approved just one suggestion to change the time

**Pull request merged**